Added Sesame Support for SparqlHttp

Maatary opened the pull request:
Support for the Sesame implementation to handle Remote repository i.e. sparql via HTTP

Maatary committed:
Added Sesame Support for SparqlHttp

Maatary commented:
Any comments ?

InTheNow commented on build.sbt:
```diff
@@ -5,6 +5,6 @@ val runExamplesStr =
   ";examples/run-main org.w3.banana.examples.IOExampleWithJena" +
   ";examples/run-main org.w3.banana.examples.IOExampleWithSesame" +
-  ";examples/run-main org.w3.banana.examples.SPARQLExampleWithJena"
+  ";examples/run-main org.w3.banana.examples.SPARQLConsumerExampleApp"
```
Please change to two example runs, in line with the other examples

InTheNow commented on misc/examples/src/main/scala/org/w3/banana/examples/SPARQLExample.scala:
```diff
@@ -4,9 +4,11 @@ import org.w3.banana._
 /* declare your dependencies as a trait with all the modules you need
  */
-trait SPARQLExampleDependencies
+trait SPARQLHttpEndPointDependencies
```
No need to change this

InTheNow commented on misc/examples/src/main/scala/org/w3/banana/examples/SPARQLExample.scala:
```diff
@@ -23,13 +25,13 @@ trait SPARQLExampleDependencies
  *   run-main org.w3.banana.examples.SPARQLExampleWithJena
  */
-trait SPARQLExample extends SPARQLExampleDependencies { self =>
+trait SPARQLConsumerExampleMixin extends SPARQLHttpEndPointDependencies {
```
ditto, keep as before

InTheNow commented on misc/examples/src/main/scala/org/w3/banana/examples/SPARQLExample.scala:
```diff
@@ -38,13 +40,14 @@ trait SPARQLExample extends SPARQLExampleDependencies { self =>
- ?other ont:influencedBy ?language .
-} LIMIT 100
-""").get
+                                PREFIX ont: <http://dbpedia.org/ontology/>
```
the format was wrong before, but still not quite correct - see https://github.com/banana-rdf/banana-rdf/blob/series/0.8.x/rdf-test-suite/jvm/src/main/scala/org/w3/banana/SparqlUpdateEngineTest.scala#L30 for an example (note .stripMargin at end)

InTheNow commented on misc/examples/src/main/scala/org/w3/banana/examples/SPARQLExample.scala:
```diff
@@ -69,5 +72,15 @@ SELECT DISTINCT ?language WHERE {
  * them. */
 
 import org.w3.banana.jena.JenaModule
+import org.w3.banana.sesame.SesameModule
```
here to the end should just be two imports, two objects

InTheNow commented on sesame/src/main/scala/org/w3/banana/sesame/SesameHttpSparqlEngine.scala:
```diff
@@ -0,0 +1,57 @@
+package org.w3.banana.sesame
```
General comments on this file, basically try and emulate the [jena version](https://github.com/banana-rdf/banana-rdf/blob/series/0.8.x/jena/src/main/scala/org/w3/banana/jena/JenaSparqlHttpEngine.scala) :

* Don't use the sesame types directly, eg rather `Sesame#Node` than `Value`
* Jena uses `QueryExecutionFactory.sparqlService` so you need to find the sesame equivalent to make the rest of the code like the jena one

InTheNow replied:
Although not the same as the sparql implementation, if you look at [SesameGraphSparqlEngine]( https://github.com/banana-rdf/banana-rdf/blob/series/0.8.x/sesame/src/main/scala/org/w3/banana/sesame/SesameGraphSparqlEngine.scala) that "looks" more like the jena implementation and what is (ideally) need here. (in fact.... almost the same.....)

Maatary replied:
Problem solved. I think we can move forward with this

Maatary commented:
Alright I have fixed up the examples, reverting it back to what it was. 

I believe  the code complies now. Please let me know. I would really like to have this commit accepted. I believe this could be useful to a lot of people. We can further in the future get the code to look more like in jena, but as i have tried i have faced few challenges that i don't have the time to figure out right now. 

Given that it does the job. Would it be possible to accept the change and then start working on embellishing after ?

antonkulaga committed:
Updated dependencies, changed the build, moved all tests to scalatest

Scala/JDK:
* remove scala 2.10
 *move to JDK 8, in order to prepare for 2.12

the n3js files are removed for the moment as they have not been maintained (hopefull can be added back later)

SBT upgrade to 0.13.12 which is faster:
* build.scala moved to build.sbt, as suggested by latest sbt manual
* improved build.sbt no longer requires new lines, looks more like normal scala code
* deletion of Alistair's very helpful but unmaintained scala.js cross-project plugin in favour of default scala.js one

Publishing file moved to build.sbt:
* needed to move folders and files due to scala.js cross projects having a different folder layout

Update dependencies
* Jena update
* move to scalatest v3.0 which supports scalajs out of the box
* upgrade to minor version improvement of Sesame. The new version rdf4j has big name changes and API changes related to their use of Java8 lambdas, so that should be for a later PR.

added ScalaStyle plugin and default xml file

Upgrade name to 0.9.0 snapshot

fixes based on Henry's and Jean-Marc's feedback

stain committed:
Updated README links to RDFOps and friends

bblfish committed:
Merge pull request #315 from antonkulaga/series/0.8.x

build configuration update, scalatest and dependencies updates

Looks good to me, and everyone has had ample time to comment.
I'll add an issue so that we can track what the "JenaFusekiSparqlTest.scala" tests are now ignored, as they may be there for a good reason, and it's a pitty to loose tests.

bblfish committed:
Merge pull request #317 from stain/patch-1

Updated README links to RDFOps and friends

stain committed:
Update README links again..

bblfish committed:
Merge pull request #319 from stain/patch-2

Update README links again..

antonkulaga committed:
Scala 2.12 support

bblfish committed:
Merge pull request #320 from antonkulaga/series/0.8.x

Scala 2.12 support. I tested this by updating an old project of mine using this https://github.com/read-write-web/bcwm/commit/3d950b1a4f04e9e72259361262053daf40c3fb81

bblfish committed:
added option to publish to private repo with ssh. Updated docs.

bblfish committed:
reverted to 0.8.x branch. See discussion
   https://gitter.im/banana-rdf/banana-rdf?at=58625f51c02c1a3959b958f3

antonkulaga committed:
Merge pull request #321 from read-write-web/PR-ssh-publication

added option to publish to private repo with ssh.

bblfish committed:
Update README.md

fixed hyperlink to maven release repo

bblfish committed:
Update README.md

add link to Ammonite Scripting wiki

bblfish committed:
publish release 0.8.4

bblfish committed:
Merge pull request #326 from read-write-web/topic/v0.8.4

this is just the first commit of [a previous pull request](https://github.com/banana-rdf/banana-rdf/pull/323), where this commit was never questioned. So in order to move on, I'll commit this myself here as it is pretty uncontroversial. If someone found a problem with moving to a new version there was a month or so to comment on it.

Maatary committed:
Added support for Sparql update over Http

Maatary committed:
Merge branch 'series/0.8.x' of https://github.com/Maatary/banana-rdf into series/0.8.x
